### DISCUSSION TITLE
Difficulty using EB on thin walls

### DISCUSSION THREAD

**Discussion by JayStandridge** [AMReX-Combustion/PeleC#643] [General]
Hi all,

I am trying to mesh the Sandia Flame A geometry:
![image](https://github.com/AMReX-Combustion/PeleC/assets/27370774/26d1b006-105a-4e8f-acaa-e7f26d5e3cb0)
I am using the extruded_triangles eb2 geometry because four triangles are all I need to match the flow due to symmetry. Due you have some recommendations on how to mesh the thin wall efficiently? I have been able to mesh the geometry, but it requires my coarse level to be excessively fine in the vast majority of the domain. When I coarsen the mesh to an appropriate level, I get the following error:

amrex::Abort::0::Failed to build required coarse EB level 4 !!!

I thought there might be an issue with the cell boundaries not aligning with the EB cuts, but that doesn't seem to fix it.
I thought it might not have enough cells across the wall to get a good stencil, but increasing my refinement doesn't seem to change the error either.

Do you have some AMR best practices or guidelines for meshing a geometry like this?

**Comment by marchdf** [comment]
@baperry2 has set up some of these kinds of flames so he might have thoughts. My question would be: is it necessary to get that geometry? Could you not have the domain start at the tip of that premixture inlet and just remove the need for EB completely? Maybe this is a naive question and I don't fully understand the issue. I guess you wouldn't get the boundary layer...

**Reply by drummerdoc** [reply]
It probably depends on the purpose of the simulations.  Often they want to capture the formation of boundary layers around the pipe as that affects standoff distance and shape of the lower parts of the flame.  If this isn't critical, it's worth thinking about simplifying the geometry to avoid the thin body.  In general, thin bodies are EB's worst nightmare.

Also, are you sure that you need a compressible solve here?  You might consider PeleLMeX, as it could be dramatically more efficient for these types of low Mach flames.

**Reply by baperry2** [reply]
If you can avoid using EB in this case that's definitely preferable so you can use Godunov instead of MOL. In my experience with similar flames, if the boundary layer in the coflow is important, you may be able to model it satisfactorily by imposing a desired profile/fluctuations at the jet exit plane rather than extending your domain upstream. I also echo Marc's point about PeleLMeX - if I recall Flame A is quite slow and the acoustic time stepping constraint would be exceptionally limiting in PeleC.

If you do need EB, I don't think there is any way of getting around a fairly severe constraint on the resolution level of your base grid - in particular the base grid should be at least a factor of sqrt(2) smaller than the thinnest wall thickness. PeleC doesn't support multicut cells (cells where an embedded boundary crosses through and divides the cell into two separate fluid domains), and to be sure of avoiding that your base grid must smaller than the thinnest solid structure in the domain.

**Reply by JayStandridge** [reply]
Thank you so much for the replies.

I am taking a look at PeleLMeX right now. We are looking for entropy violations due to the mechanism, so the specifics of the flame aren't terribly important, but we are trying to verify against flame A data. In order to get results efficiently, I've been running with a much thicker tube. I tossed it on my universities HPRC cluster with 256 cores for two days and got about 3 milliseconds of data (~4500 iterations) which is pretty abysmal. I am only seeing a maximum mach of 0.025, so PeleLM might worth looking at.

@baperry2 When you mention the base grid do you mean the coarsest level or the finest level? The way the input files is structured had me thinking that the coarse grid is the base, but it appears the code builds the finest grid first and then coarsens. Could you also expand on why Godunov is preferable to MOL? This is not obvious to me. Could I also ask why Godunov is implemented? In my classes I've been taught that approximate riemann solvers are significantly faster while being just as accurate as Godunov solvers.

Thank you all for your time

**Reply by drummerdoc** [reply]
The "base grid" is the coarsest, amr level = 0

The EB is built from the finest grid down the coarsest (and for LMeX, we would also like it to coarsen even further in order to maximize the depth of the mg V cycles).

**Reply by baperry2** [reply]
BTW, building EB from finest to coarsest is done to ensure that the EB on coarse levels is consistent with the EB on the finer levels.

Regarding the numerical methods, the short answer is that you can use a larger timestep and its less dissipative. See the discussion here: https://amrex-combustion.github.io/PeleC/Algorithms.html#comparison-of-ppm-and-mol-for-the-decay-of-homogeneous-isotropic-turbulence
